doc/adr/0007-api-service.md

```diff
@@ -0,0 +1,106 @@
+
+## Decision
+
+The proposal consists of extracting the API related to applications, which we will call `wego-api`, execute it within the cluster through using deployment and expose it through a service. The wego-api will be the only entity that will be able to interact directly with the Kubernetes API. In other words, the wego-api will expose the necessary options to interact with wego applications, abstracting the logic that interacts with the Kubernetes API.
+
+To illustrate, you could say that the `wego` command line could tell the wego-api "Add this application" and it, in turn, would tell Kubernetes "Apply these manifest files. "
+
+For security reasons, permissions will be implemented through RBAC so that the wego-api only has access to the resources it needs.
+
+With this proposal, the use of CLI options would still be required, however, it would be an intermediate step to eventually do
+a 100% migration into the cluster.
+
+Specifically, the proposed changes are as follows:
+
+### Create wego-api docker image and update authentication method.
+
+For the wego-api to run within the cluster, a docker image will need to exist that can run it.
+
+The part that needs to be run in this image is the HTTP server which is initialized in the `wego ui run` command.
+
+To continue communicating with the Kubernetes API, it will be required to change the authentication method to one that allows communication from within the cluster called in-cluster client configuration.
+
+Kubectl and flux will be added to this docker image as well.
+
+### Generate RBAC-based permissions to the wego-api and apply them when installing the GitOps runtime.
+
+Since the wego-api needs to access resources within the cluster, RBAC permissions will be applied so that there are no security gaps. In consideration that the functionality of the wego-api will be specified only in actions within the namespace where the GitOps runtime is executed, the Role and RoleBinding resources will be used. Since the scope of this type of resource is by namespace. 
+
+The resources that will be required are:
+
+- ServiceAccount
+- Deployment
+- Role
+- RoleBinding
+- Service
+
+The specific permissions for the Role resource will be:
+
+```
+resources: ["apps.wego.weave.works"]
+verbs: ["get","create","list"]
+
+resources: ["secret"]
+verbs: ["get","create"]
+
+resources: ["kustomizations.kustomize.toolkit.fluxcd.io"]
+verbs: ["get","create"]
+
+resources: ["helmreleases.helm.toolkit.fluxcd.io"]
+verbs: ["get","create"]
+
+resources: ["helmrepositories.source.toolkit.fluxcd.io"]
+verbs: ["get","create"]
+```
+
+These resources will be applied when installing the GitOps engine.
+
+### Enable access to the wego-api from the client-side.
+
+To enable access to the `wego-api` from the client machine, we will use `kubectl` by running this command: `kubectl -n NAMESPACE port-forward deployment/wego-api :8283`. An example of the output would be: 
```

Comment:
The user should have the option of the local port to expose.

Reply:
Is the user expected to open the port prior to running the command(s)? Or is the port forwarding done on their behalf?

Reply:
It's worth mentioning that another ADR will cover API authentication/authorization

Reply:
We could add the option that the user can choose the port but we might not be getting any additional benefit. The way this command works without specifying local port(`LOCAL_PORT:EXTERNAL_PORT` like `:8383`), is that it will forward a connection to random and not-used local port to the port on the wego-api pod. More details here. https://kubernetes.io/docs/tasks/access-application-cluster/port-forward-access-application-cluster/#let-kubectl-choose-local-port. Actually I don't think we even need a service, we might just need a deployment 🤔

Reply:
Not sure about the API authentication/authorization. `kubectl port-forward` does that automatically. Do you mean like an extra layer where only wego CLI or UI can access to it ?